src/releases/semver/version.rs

```diff
@@ -0,0 +1,140 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::step::StepError;
+
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub(crate) struct Version {
+    pub(crate) major: u64,
+    pub(crate) minor: u64,
+    pub(crate) patch: u64,
+    pub(crate) pre: Option<Prerelease>,
+}
+
+impl Ord for Version {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.major
+            .cmp(&other.major)
+            .then(self.minor.cmp(&other.minor))
+            .then(self.patch.cmp(&other.patch))
+            .then({
+                match (&self.pre, &other.pre) {
+                    (Some(pre), Some(other_pre)) => pre.cmp(other_pre),
+                    (Some(_), None) => std::cmp::Ordering::Less,
+                    (None, Some(_)) => std::cmp::Ordering::Greater,
+                    (None, None) => std::cmp::Ordering::Equal,
+                }
+            })
+    }
+}
+
+impl PartialOrd for Version {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl FromStr for Version {
+    type Err = StepError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (version, pre) = s
+            .split_once('-')
+            .map_or((s, None), |(version, pre)| (version, Some(pre)));
+        let version_parts = version
+            .split('.')
+            .map(|part| {
+                part.parse::<u64>()
+                    .map_err(|err| StepError::InvalidSemanticVersion(err.to_string()))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        if version_parts.len() != 3 {
+            return Err(StepError::InvalidSemanticVersion(
+                "Version must have 3 parts".to_string(),
```

Comment:
To differentiate from potentially meaning "at least 3"

```suggestion
                "Version must have exactly 3 parts".to_string(),
```